patches/api/0360-Don-t-load-plugins-prefixed-with-a.patch

```diff
@@ -0,0 +1,18 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Noah van der Aa <ndvdaa@gmail.com>
+Date: Sat, 22 Jan 2022 16:35:44 +0100
+Subject: [PATCH] Don't load plugins prefixed with a .
+
+
+diff --git a/src/main/java/org/bukkit/plugin/SimplePluginManager.java b/src/main/java/org/bukkit/plugin/SimplePluginManager.java
+index 1366496271c4c7f72d1e5f990e51775b1c371f99..51bfae5f96b048482651cbbd2eade6beacbae80d 100644
+--- a/src/main/java/org/bukkit/plugin/SimplePluginManager.java
++++ b/src/main/java/org/bukkit/plugin/SimplePluginManager.java
+@@ -142,6 +142,7 @@ public final class SimplePluginManager implements PluginManager {
+         final List<File> pluginJars = new ArrayList<>(java.util.Arrays.asList(directory.listFiles()));
+         pluginJars.addAll(extraPluginJars);
+         for (File file : pluginJars) {
++            if (file.getName().startsWith(".")) continue; // Don't load plugin if the file name starts with a .
```

Comment:
This should also check if it isn’t an extra plugin jar

Reply:
It _can_ but what if it loads the dot file first?

Reply:
> It _can_ but what if it loads the dot file first?

I assume this is likely to happen because of alphabetical ordering.
So I don't believe we can check for the extra plugin jar .. that is unless we loop thru each jar to see if the plugin.yml name value is the same but that seems too extra to achieve the desired outcome of the PR :)
What do you think?

Reply:
Oh, far out, I had the complete wrong perception of what an extra plugin JAR was. I apologize.